### DISCUSSION TITLE
Stop animation

### DISCUSSION THREAD

**Discussion by REOREV** [google/model-viewer#2482] [Q&A] [answered]
Hello,

I try to stop an animation at the end of this one but actually I have 2 issues: 
either the animation is looping or the animation back to the begining of this animation.

Thanks for your help

**Comment by elalish** [comment]
You can call `pause()` after a timeout if you want it to play once. Is that what you want?

**Reply by adrianposteuca** [reply]
Is there a function onAnimationEnd?

**Reply by elalish** [reply]
No, but you can check `currentTime` in a RAF callback.

**Reply by elalish** [reply]
Wait, I take it back: https://modelviewer.dev/docs/index.html#entrydocs-animation-events-finished

**Reply by adrianposteuca** [reply]
I am a newbie in JS  so this is the right way to do it?
I just want to make visible a button on the animation end.

  modelViewer.animation.finished ('finish',()=>{
        let img = document.querySelector("#theImage");
        img.style.visibility = 'visible';   
      });

**Reply by elalish** [reply]
You'll need an event listener: `modelViewer.addEventListener('finished', () => {...});`

**Reply by adrianposteuca** [reply]
I have tried but it seems like is not working:
I just add a window. alert ("end") under it and it doesn't call the function... wired!

I ve tried this syntax too: modelViewer.finished = Function (){ ....}

Any ideas?

**Reply by elalish** [reply]
The syntax you put there is definitely not how our API works. Can you share a link?

**Reply by Cebolitoss** [reply]
I'm having the same issue, could you fix it?

At the first time I hit the animation button, it runs in loop, at the next hits, it stop, but sometimes it stop at the last frame of the animation, and sometimes it backs to the first frame and stop there, in the first one.

And when I have two animations, they run together and it makes just a mess

**Reply by bigrig2212** [reply]
Hi - the finished event listener doesn't seem to be working. here's my function. See anything fishy there? 

```
CreateNPCSimple.prototype.changeBuilderAnimation = function(animation, id) {
    let modelViewer = document.querySelector('#'+id);
    modelViewer.animationName = animation;
    //if animation is death, pause all animations at the end of one loop
    if (animation == 'death') {
        this.mlog.log('CreateNPCSimple-changeBuilderAnimation - DEATH', {}, 6); 
        modelViewer.addEventListener('finished', () => {
            this.mlog.log('CreateNPCSimple-changeBuilderAnimation - DEATH - ANIM FINISH', {}, 6); 
            modelViewer.pause();
        });
    } else { 
        modelViewer.play();
    }
}
```

**Reply by elalish** [reply]
Not sure about the event, but have you tried a simpler approach? `modelViewer.play({repetitions: 1})` https://modelviewer.dev/docs/#entrydocs-animation-methods-play

**Reply by bigrig2212** [reply, marked as answer]
Beautiful. There was one more piece needed to get it to work, but it's great now. 
```
modelViewer.animationName = animation;
 await modelViewer.updateComplete;
 modelViewer.play({repetitions: 1})
```
Thanks for your helo.
